pipeline_components/aiplatform/aiplatform/model_batch_predict/component.py

```diff
@@ -0,0 +1,139 @@
+    )
+    from google.cloud.aiplatform_v1beta1.types.job_state import JobState
+
+    api_endpoint = f"{project_location}-aiplatform.googleapis.com"
+
+    alert_config = None
+    if alert_email_addresses is not None:
+        alert_config = ModelMonitoringAlertConfig(
+            email_alert_config=ModelMonitoringAlertConfig.EmailAlertConfig(
+                user_emails=[alert_email_addresses]
+            )
+        )
+
+    skew_detection_config = (
+        ModelMonitoringObjectiveConfig.TrainingPredictionSkewDetectionConfig(
+            default_skew_threshold=ThresholdConfig(value=0.001),
+        )
+    )
+
+    job_request = BatchPredictionJob(
+        display_name=job_display_name,
+        model=model.metadata["resourceName"],
+        input_config=BatchPredictionJob.InputConfig(
+            instances_format="bigquery",
+            bigquery_source=BigQuerySource(input_uri=bigquery_source_input_uri),
+        ),
+        output_config=BatchPredictionJob.OutputConfig(
+            predictions_format="bigquery",
+            bigquery_destination=BigQueryDestination(
+                output_uri=bigquery_destination_output_uri
+            ),
+        ),
+        dedicated_resources=BatchDedicatedResources(
+            machine_spec=MachineSpec(machine_type=machine_type),
+            starting_replica_count=starting_replica_count,
+            max_replica_count=max_replica_count,
+        ),
+        model_monitoring_config=ModelMonitoringConfig(
+            alert_config=alert_config,
+            objective_configs=[
+                ModelMonitoringObjectiveConfig(
+                    training_dataset=ModelMonitoringObjectiveConfig.TrainingDataset(
+                        bigquery_source=BigQuerySource(
+                            input_uri=bigquery_source_training_uri
+                        ),
+                    ),
+                    training_prediction_skew_detection_config=skew_detection_config,
+                )
+            ],
+        ),
+    )
+
+    logging.info(f"Submitting batch prediction job: {job_display_name}")
+    client = JobServiceClient(client_options={"api_endpoint": api_endpoint})
+    job_response = client.create_batch_prediction_job(
+        parent=f"projects/{project_id}/locations/{project_location}",
+        batch_prediction_job=job_request,
+    )
+    logging.info(f"Submitted batch prediction job: {job_response.name}")
+
+    states = [
+        JobState.JOB_STATE_SUCCEEDED,
+        JobState.JOB_STATE_FAILED,
+        JobState.JOB_STATE_CANCELLED,
+        JobState.JOB_STATE_EXPIRED,
+    ]
+    job_status_request = GetBatchPredictionJobRequest(name=job_response.name)
+    state = client.get_batch_prediction_job(request=job_status_request).state
+
+    while state not in states:
+        logging.info(f"Batch prediction job state: {state}, waiting for 10s...")
+        time.sleep(10)
+        state = client.get_batch_prediction_job(request=job_status_request).state
```

Comment:
I think you can just set it running and then return the protobuf as a string - see https://github.com/kubeflow/pipelines/tree/master/components/google-cloud/google_cloud_pipeline_components/proto